kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/utils/scheduling/Task.kt

```diff
@@ -121,23 +133,29 @@ public open class Task(
             logger.error(t) { "Error running scheduled callback." }
         }
 
-        removeFromParent()
+        if (!repeat) {
+            removeFromParent()
+        }
     }
 
     /** Stop waiting and don't execute. **/
     public fun cancel() {
         job?.cancel()
         job = null
 
-        removeFromParent()
+        if (!repeat) {
+            removeFromParent()
+        }
```

Comment:
There's no point in adding this conditional here, it should _always_ be removed as it stops running at this point.